lua/telescope/utils.lua

```diff
@@ -170,6 +170,63 @@ utils.filter_symbols = function(results, opts, post_filter)
   end
 end
 
+utils.path_filename_first = function(path, reverse_directories)
```

Comment:
all functions that are only used in the file should be local, if they are used in tests as well export them with `__` we do want to guarantee a stable interface for those thats why they need to be maximum `__`, but local would be ideal.

`transform_path` is the only interface that extension devs should use